src/pkg/task/manager.go

```diff
@@ -0,0 +1,53 @@
+// Copyright Project Harbor Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//    http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package task
+
+import (
+	"context"
+
+	"github.com/goharbor/harbor/src/lib/q"
+)
+
+// Manager provides an execution-task model to abstract the interactive with jobservice.
+// All of the operations with jobservice should be delegated by this manager
+type Manager interface {
+	// CreateExecution creates an execution. The "typee" is used to specify the type(execution, scan, etc.)
+	// of the execution, and the "extraAttrs" can be used to set the customized attributes
+	CreateExecution(ctx context.Context, typee, trigger string, extraAttrs ...map[string]interface{}) (id int64, err error)
+	// UpdateExecutionStatus updates the status of the execution.
+	// In most cases, the execution status can be calculated from the referenced tasks automatically.
+	// When the execution contains no tasks or failed to create tasks, the status should be set manually
+	UpdateExecutionStatus(ctx context.Context, id int64, status, message string) (err error)
```

Comment:
Support both options. For simplification, maybe we only need to support one. Calculating the data from the sub-tasks?